sample_project/Assets/SampleViewer/Samples/Identify/Scripts/Identify.cs

```diff
@@ -0,0 +1,430 @@
+using UnityEngine.UI;
+
+public class Identify : MonoBehaviour
+{
+    [Header("Identify Variables")]
+    [SerializeField] private ArcGISMapComponent arcGISMapComponent;
+    private ArcGIS3DObjectSceneLayer buildingLayer;
+    [SerializeField] private Transform buildingListContentContainer;
+    private List<GameObject> buildingListItems = new List<GameObject>();
+    [SerializeField] private GameObject buildingToggle;
+    [SerializeField] private Transform contentContainer;
+    private List<GameObject> ListItems = new List<GameObject>();
+    [SerializeField] private GameObject markerGO;
+    [SerializeField] private float maxRayLength = 10000000;
+    [SerializeField] private GameObject scrollViewItem;
+    [SerializeField] private Color selectColor;
+    [SerializeField] private Material selectMaterial;
+
+    [Header("UI Elements")]
+    [SerializeField] private GameObject buildingsView;
+    [SerializeField] private Button changeViewButton;
+    [SerializeField] private Button decreaseResult;
+    [SerializeField] private Button increaseResult;
+    [SerializeField] private GameObject menuBar;
+    [SerializeField] private TextMeshProUGUI resultAmount;
+    [SerializeField] private GameObject results;
+    [SerializeField] private Button showResultsButton;
+    [SerializeField] private TextMeshProUGUI totalNumberOfBuildingsText;
+
+    // Select Variables
+    private ArcGISImmutableCollection<ArcGISIdentifyLayerResult> identifyLayerResults;
+    private ulong resultsLength;
+    private float selectedID;
+    [HideInInspector] public ulong SelectedResult = 0;
+
+    [Header("Thread Safety")]
+    [SerializeField] private float identifyLayersTimeoutSeconds = 5f;
+    [SerializeField] private bool cancelOnTimeout = true; // If true, call Cancel() on the future when timing out.
+    [SerializeField] private bool verboseResultLogging = true; // Dump full attribute JSON-ish payload; disable for brevity.
+    private readonly Queue<System.Action> mainThreadActions = new(); // Actions queued from callback
+    private Coroutine activeIdentifyLayersCoroutine;
+
+    private IEnumerator CallbackTimeoutWatcher(ArcGISFuture<ArcGISImmutableCollection<ArcGISIdentifyLayerResult>> future, float start, System.Action onTimeout)
+    {
+        while (!future.IsDone())
+        {
+            if (identifyLayersTimeoutSeconds > 0 && (Time.realtimeSinceStartup - start) > identifyLayersTimeoutSeconds)
+            {
+                Debug.LogWarning("IdentifyLayersAsync (callback) timed out.");
+                if (cancelOnTimeout)
+                {
+                    future.Cancel();
+                }
+                onTimeout?.Invoke();
+                yield break;
+            }
+            yield return null;
+        }
+    }
+
+    //Filter out incorrect date format. See known issue (BUG-000181006): https://developers.arcgis.com/unity/release-notes/release-notes-2-2-0/.
+    private bool DateTimeCheck(object AttributeValue)
```

Comment:
Can we rename this function? Name indicates that returning true means the value passed the check. Maybe `DateTimeIsInvalid`?